test/parallel/test-stream-pipe-objectmode-to-non-objectmode.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('node:assert');
+const { Readable, Transform, Writable } = require('node:stream');
+
```

Comment:
As a new test, it would be helpful to include comments that describe what each individual block is testing here, or, perhaps use `node:test` and give each block a descriptive name.

Reply:
Thanks for the suggestion! I will try to add some helpful comments to make the test more clear 👍